--- acpi-support-0.133.orig/asus-wireless.sh
+++ acpi-support-0.133/asus-wireless.sh
@@ -5,4 +5,18 @@
 
 . /usr/share/acpi-support/state-funcs
 
-toggleAllWirelessStates
+VENDOR=`cat /sys/class/dmi/id/sys_vendor | tr [:upper:] [:lower:] | cut -c1-4`
+test $VENDOR = "asus" || exit 0
+
+if [ "$1" = "" ] ; then
+	toggleAllWirelessStates;
+elif ( isAnyWirelessPoweredOn ) ; then
+	if [ "$1" = "off" ] ; then
+		toggleAllWirelessStates;
+	fi
+else
+	if [ "$1" = "on" ] ; then
+		toggleAllWirelessStates;
+	fi
+fi
+
