--- acpi-support/lib/screenblank
+++ acpi-support/lib/screenblank
@@ -1,7 +1,7 @@
 if [ `pidof xscreensaver` ]; then
-	su $user -s /bin/sh -c "(xscreensaver-command -throttle)"
+	su $user -s /bin/sh -c "(xscreensaver-command -throttle&)"
 		if [ x$LOCK_SCREEN = xtrue ]; then	
-		su $user -s /bin/sh -c "(xscreensaver-command -lock)"
+		su $user -s /bin/sh -c "(xscreensaver-command -lock&)"
 	fi
 elif [ `pidof dcopserver` ]; then
 	if [ x$LOCK_SCREEN = xtrue ]; then
@@ -16,13 +16,13 @@ fi
 
 case "$DISPLAY_DPMS" in
   xset)
-su $user -s /bin/sh -c "xset dpms force off"
+su $user -s /bin/sh -c "xset dpms force off&"
 	;;
   xrandr)
-	su $user -s /bin/sh -c "xrandr --output LVDS --off"
+	su $user -s /bin/sh -c "xrandr --output LVDS --off&"
 	;;
   vbetool)
-	/usr/sbin/vbetool dpms off
+	/usr/sbin/vbetool dpms off &
 	;;
 esac
 
