--- acpi-support/lib/policy-funcs	2009-12-07 20:18:55.643070790 +0100
+++ acpi-support/lib/policy-funcs	2009-12-07 20:19:15.489729366 +0100
@@ -26,7 +26,7 @@
 	
 	for p in $(pidof kded4); do
 		test -r /proc/$p/environ || continue
-		local DBUS_SESS=$(cat /proc/$p/environ | grep -z "DBUS_SESSION_BUS_ADDRESS=")
+		local DBUS_SESS=$(cat /proc/$p/environ | grep -a -z "DBUS_SESSION_BUS_ADDRESS=")
 		test "$DBUS_SESS" != "" || continue
 		(su - $(ps -o user= $p) -s /bin/sh -c "$DBUS_SESS dbus-send --print-reply --dest=org.kde.kded /kded org.kde.kded.loadedModules" | grep -q powerdevil) && return 0
 	done
