--- acpi-support-0.122.orig/lib/screenblank
+++ acpi-support-0.122/lib/screenblank
@@ -4,7 +4,14 @@
 		su $user -c "(xscreensaver-command -lock)"
 	fi
 elif [ `pidof dcopserver` ]; then
-	dcop kdesktop KScreensaverIface lock
+	if [ x$LOCK_SCREEN = xtrue ]; then
+		avail_sessions=`dcop --all-users --list-sessions | grep '.DCOP.*__0'`
+    		# send the lock command to all sessions
+    		for session in $avail_sessions; do
+		    	# /dev/null because dcop warns if it can't connect to X (this is normal!)
+		    	dcop --session "$session" --all-users kdesktop KScreensaverIface lock > /dev/null 2>&1
+    		done
+	fi
 fi
 
 xset dpms force off
