diff -ruN acpi-support.orig/sleep.sh acpi-support/sleep.sh
--- acpi-support.orig/sleep.sh	2009-11-22 12:27:18.000000000 +0100
+++ acpi-support/sleep.sh	2009-11-22 12:27:41.000000000 +0100
@@ -1,5 +1,4 @@
-#!/bin/bash
-# TODO:  Change above to /bin/sh
+#!/bin/sh
 
 # This script HANDLES the sleep button (does not TRANSLATE it). It is part
 # of the *suspend* side of acpi-support, not the special keys translation
