diff -urN xl2tpd-1.3.6+dfsg.orig/network.c xl2tpd-1.3.6+dfsg.untangle/network.c
--- xl2tpd-1.3.6+dfsg.orig/network.c	2014-01-15 13:58:37.000000000 -0700
+++ xl2tpd-1.3.6+dfsg.untangle/network.c	2014-09-14 07:37:19.000000000 -0700
@@ -280,7 +280,7 @@
 void udp_xmit (struct buffer *buf, struct tunnel *t)
 {
     struct cmsghdr *cmsg;
-    char cbuf[CMSG_SPACE(sizeof (unsigned int) + sizeof (struct in_pktinfo))];
+    char cbuf[CMSG_SPACE(sizeof (unsigned int) + CMSG_SPACE(sizeof (struct in_pktinfo)))];
     unsigned int *refp;
     struct msghdr msgh;
     int err;
@@ -292,6 +292,7 @@
      * OKAY, now send a packet with the right SAref values.
      */
     memset(&msgh, 0, sizeof(struct msghdr));
+    memset(cbuf, 0, sizeof(cbuf));
 
     cmsg = NULL;
     msgh.msg_control = cbuf;
@@ -310,7 +311,7 @@
 	refp = (unsigned int *)CMSG_DATA(cmsg);
 	*refp = t->refhim;
 	
-	finallen = cmsg->cmsg_len;
+	finallen = CMSG_SPACE(sizeof (unsigned int));
     }
     
     if (t->my_addr.ipi_addr.s_addr){
@@ -329,7 +330,7 @@
 	pktinfo = (struct in_pktinfo*) CMSG_DATA(cmsg);
 	*pktinfo = t->my_addr;
 	
-	finallen += cmsg->cmsg_len;
+	finallen += CMSG_SPACE(sizeof (struct in_pktinfo));
     }
     
     msgh.msg_controllen = finallen;
