Description: miniupnpd/upnpredirect.c:  check inet_aton() return
Author: Thomas Bernard <miniupnp@free.fr>
Origin: upstream, https://github.com/miniupnp/miniupnp/commit/7c91c4e933e96b913b72685d093126d282b87db6.patch
Date: Tue, 9 Dec 2014 10:53:52 +0100
Bug-Debian: https://bugs.debian.org/772644
Last-Update: 2014-12-09

diff --git a/upnpredirect.c b/upnpredirect.c
index 5a50c6d..28560ab 100644
--- a/upnpredirect.c
+++ b/upnpredirect.c
@@ -268,8 +268,8 @@ upnp_redirect(const char * rhost, unsigned short eport,
 	unsigned int timestamp;
 
 	proto = proto_atoi(protocol);
-	if(inet_aton(iaddr, &address) < 0) {
-		syslog(LOG_ERR, "inet_aton(%s) : %m", iaddr);
+	if(inet_aton(iaddr, &address) <= 0) {
+		syslog(LOG_ERR, "inet_aton(%s) FAILED", iaddr);
 		return -1;
 	}
 
